src/Routing/Route/DashedRoute.php

```diff
@@ -56,11 +56,12 @@ protected function _camelizePlugin($plugin)
      * camelBacked form.
      *
      * @param string $url The URL to parse
+     * @param string $method The HTTP method.
      * @return array|false An array of request parameters, or false on failure.
      */
-    public function parse($url)
+    public function parse($url, $method = '')
```

Comment:
But that is only if it is not an interface, right? Because interfaces would always make your code fail, whereas extension and stuff works due to the default value.